sklearn/tests/test_naive_bayes.py

```diff
@@ -438,6 +445,45 @@ def test_discretenb_coef_intercept_shape(cls):
     assert clf.intercept_.shape == (1,)
 
 
+@pytest.mark.parametrize('cls', [MultinomialNB, ComplementNB, BernoulliNB,
+                                 CategoricalNB])
+@pytest.mark.parametrize('use_partial_fit', [False, True])
+@pytest.mark.parametrize('num_samples,expected_first_axis_length',
+                         [(3, 2), (2, 1)])
+def test_discretenb_degenerate_one_class_case(
+        cls,
+        use_partial_fit,
+        num_samples,
+        expected_first_axis_length,
+):
+    # Most array attributes of a discrete naive Bayes classifier should have
+    #     a first-axis length equal to the number of classes. Exceptions
+    #     include: ComplementNB.feature_all_, CategoricalNB.n_categories_.
+    # Confirm that this is the case for binary problems and degenerate
+    #     one-class problems when fitting with `fit` or `partial_fit`.
+    # Non-regression test for handling degenerate one-class case:
+    #     https://github.com/scikit-learn/scikit-learn/issues/18974
+
+    X = [[1, 0, 0], [0, 1, 0], [0, 0, 1]][:num_samples]
+    y = [1, 1, 2][:num_samples]
+    clf = cls()
```

Comment:
```suggestion
    clf = NaiveBayes()
```

Reply:
Will do locally along with above and line 448

Reply:
Done in 4c5f8d2